core/modules/path/path.module

```diff
@@ -405,8 +732,36 @@ function path_entity_update(Entity $entity) {
     $langcode = isset($entity->langcode) ? $entity->langcode : LANGUAGE_NONE;
     $uri = $entity->uri();
     $path = $entity->path;
+    $path_original = $entity->original->path;
     $path['source'] = $uri['path'];
 
+    //save a redirect if the path is changed on node save.
```

Comment:
This applies to more than just nodes. And some minor formatting. Fix to:

`// Save a redirect if the path is changed.`